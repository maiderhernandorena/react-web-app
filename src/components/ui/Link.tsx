import React from 'react';

type LinkProps = {
  link: string
  message?: string
}

const Link: React.FC<LinkProps> = ({ link, message }) => {
  return (
      <a
        className="App-link"
        href={link}
      >
        {message || link}
      </a>
  );
}


export default Link;
