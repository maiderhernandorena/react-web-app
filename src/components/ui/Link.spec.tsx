import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Link from './Link';

describe('Link Component', () => {

  describe('Snapshots', () => {
    it('Should render with only link', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Link link='google.com' />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render with link and message', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Link link='google.com' message='message'/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});