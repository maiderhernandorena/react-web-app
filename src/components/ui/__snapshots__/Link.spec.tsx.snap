// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Snapshots Should render with link and message 1`] = `
<a
  className="App-link"
  href="google.com"
>
  message
</a>
`;

exports[`Link Component Snapshots Should render with only link 1`] = `
<a
  className="App-link"
  href="google.com"
>
  google.com
</a>
`;
